index.html

```diff
@@ -530,27 +590,65 @@ <h3>Script Event</h3>
               </li>
-              <li>The <code>p</code> element MUST have an <code>xml:lang</code> attribute corresponding to the language of the <a>Text</a> object.
-              <div class="note">Within a <code>div</code> representing a <a>Script Event</a>, the order of <code>p</code> is one relevant. The <code>xml:lang</code> value can be used to determine which paragraph corresponds to the <a>Main Text</a> and which one corresponds to the <a>Contextual Text</a>. Text content in a language that does not match the <a>Primary Language</a> of the <a>DAPT Script</a> are <a>Contextual Text</a>.</div>
+              <li>The <code>&lt;p&gt;</code> element MUST have one <a>Text Language Source</a> property.</li>
+              <li>The <code>&lt;p&gt;</code> element SHOULD have an <code>xml:lang</code> attribute corresponding to the language of the <a>Text</a> object.
+                <p class="note">If a <code>&lt;p&gt;</code> element omits the <code>xml:lang</code> attribute then its computed language
+                is derived by inheritance from its parent element, and so forth up to the root <code>&lt;tt&gt;</code> element,
+                which is required to set the <a>Primary Language</a> via its <code>xml:lang</code> attribute.
+                Care should be taken if changing the <a>Primary Language</a> of a <a>Document Instance</a> in case
+                doing so affects descendant elements unexpectedly.
+                Authors can mitigate this risk by explicitly setting <code>xml:lang</code> on all <code>&lt;p&gt;</code> elements.</p>
               </li>
           <pre class="example">
 &lt;div xml:id=&quot;event_3&quot;
      begin=&quot;9663f&quot; end=&quot;9682f&quot; 
      style=&quot;style_a&quot;
      ttm:agent=&quot;character_3&quot;&gt;
-  &lt;p xml:lang=&quot;pt-BR&quot; &gt;Voc&#xea; vai ter.&lt;/p&gt;
-  &lt;p xml:lang=&quot;fr&quot; ttm:role=&quot;x-context&quot;&gt;Bah, il arrive.&lt;/p&gt;
+  &lt;p xml:lang=&quot;pt-BR&quot; daptm:tls=&quot;original&quot; &gt;Voc&#xea; vai ter.&lt;/p&gt;
+  &lt;p xml:lang=&quot;fr&quot; daptm:tls=&quot;translation&quot; &gt;Bah, il arrive.&lt;/p&gt;
 &lt;/div&gt;
           </pre>
         </section>
 
+        <section>
+          <h4>Text Language Source</h4>
+          <p>The <dfn>Text Language Source</dfn> property is an annotation indicating whether a <a>Text</a> object is
+            in the same language as the relevant part of the <a>Related Media Object</a>'s
+            language (original), or if it is a representation in another language (translation):</p>
+          <ul>
+            <li><dfn>Original</dfn> - the <a>Text</a> is in the same language as the dialogue that it represents in the original programme audio,
+              or it represents non-dialogue sound and is in the <a>Primary Language</a> of the <a>Document Instance</a>.</li>
```

Comment:
Non-dialog sounds don't have a language, do they? Did you mean "on screen text" but in that case, it may also not be in primary language.

Reply:
Yes, text representations of non-dialogue sounds have a language. If the text is "clapping" that's an English language description of a non-dialogue sound. Agreed the sound itself doesn't have a language, but it's unusual to try to write the sound down phonetically.

Reply:
Oh, I see. Can we also add on-screen too?

Reply:
Done in 3cfe8db. Please review this change.